### PR TITLE
77 upcoming interviews for mentor

### PR DESCRIPTION
**Pull request by sharifyy**

### CODE REVIEW

File: src/main/kotlin/com/linchpino/core/service/InterviewService.kt [hunk 1]
```diff
@@ -72,4 +79,14 @@ class InterviewService(
             entity.jobSeekerAccount?.email,
         )
     }
+
+    @Transactional(readOnly = true)
+    fun upcomingInterviews(authentication: Authentication, page: Pageable): Page<InterviewListResponse> {
+        val email = when (authentication) {
+            is JwtAuthenticationToken -> authentication.name
+            is BearerTokenAuthentication -> (authentication.principal as OAuth2IntrospectionAuthenticatedPrincipal).name
+            else -> throw UnsupportedOperationException("Authentication type not supported")
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/service/InterviewService.kt:88]: Are we have any test case to getting UnsupportedOperationException("Authentication type not supported") ?

**Review by m-tilab — COMMENTED** [review]
Do we have a test case for getting 401 and 403 status?

File: src/main/kotlin/com/linchpino/core/controller/InterviewController.kt [hunk 2]
```diff
@@ -31,4 +36,18 @@ class InterviewController(private val service: InterviewService) {
...
+
+    @Operation(summary = "Return page of upcoming interviews for authenticated mentor")
+    @ApiResponses(
+        value = [
+            ApiResponse(responseCode = "200", description = "Upcoming interviews fetched successfully"),
+            ApiResponse(responseCode = "401", description = "User is not authenticated"),
+            ApiResponse(responseCode = "403", description = "Authenticated user is not MENTOR")
+        ]
+    )
+    @GetMapping("/upcoming")
```
Comment (m-tilab) [inline src/main/kotlin/com/linchpino/core/controller/InterviewController.kt]: Do you have any idea for the url mapping when a job seeker wants to get list of the upcoming interviews ?

File: src/main/kotlin/com/linchpino/core/service/InterviewService.kt (hunk 1 above)
Comment (sharifyy) [reply src/main/kotlin/com/linchpino/core/service/InterviewService.kt:88]: It should never happen, but the when expression is not exhaustive in this case so we need to throw an exception.

File: src/main/kotlin/com/linchpino/core/controller/InterviewController.kt (hunk 2 above)
Comment (sharifyy) [reply src/main/kotlin/com/linchpino/core/controller/InterviewController.kt]: Should we add mentors before upcoming? /mentors/upcoming?
I am open to any suggestions.

Comment (m-tilab) [reply src/main/kotlin/com/linchpino/core/controller/InterviewController.kt]: I think it a good Idea same as this: api/interviews/mentors/upcoming

**Review by m-tilab — APPROVED** [review]

**Pull request merged**